int16 num_point
int16 last_point
